addons/docs/src/frameworks/vue/preset.ts

```diff
@@ -1,12 +1,24 @@
 export function webpackFinal(webpackConfig: any = {}, options: any = {}) {
```

Comment:
Thank you very much for taking the time to review ! You are definitely right, I also took the liberty of specifying two interfaces :
* `Preset` for a single preset type definition
* `PresetOptions` for the actual `options`

I find it a bit easier to read this way but please let me know if it feels wrong to you !

Also, is there any reason we should prefer/avoid defining these interfaces into a separate `types.ts` file and then import them into the `preset.ts` file ?

Reply:
> Also, is there any reason we should prefer/avoid defining these interfaces in a separate `types.ts` file and then import them into the `preset.ts` file ?

I'm not sure there are strict rules about typings in this monorepo, but it seems "where to put typings?" is up to authors (example: [`types.ts`](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/angular/types.ts) / [in file](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/angular/types.ts)).

IMO core libs should have exported these typings, so commenting these with such as `FIXME: move this into framework agnostic package` would be good for future code?

Reply:
I took a quick look at the other `preset.ts` files and I think the [svelte one](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/svelte/preset.ts#L3-L4) got it right by importing both:
* `Configuration` from `webpack`
*  `Options` from `@storybook/core-common`, which includes the `StorybookConfigOptions` we were looking for.

The `presetsList` property is however set as optional, which - I guess - requires additional check before iterating the `options.presetsList` within the `webpackFinal` function. I'll commit changes if it still looks good to you.

One more _maybe stupid_ question: although I did not have the chance to check it out, I assume the issue is also occuring with Vue 3 since [vue3's `preset.ts` file](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/vue3/preset.ts) seems to be a simple copy of the vue one. Does this imply a whole new pull request or may this be addressed in this one ?

Reply:
> I took a quick look at the other `preset.ts` files and I think the [svelte one](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/svelte/preset.ts#L3-L4) got it right by importing both:
> 
>     * `Configuration` from `webpack`
> 
>     * `Options` from `@storybook/core-common`, which includes the `StorybookConfigOptions` we were looking for.
> 
> 
> The `presetsList` property is however set as optional, which - I guess - requires additional check before iterating the `options.presetsList` within the `webpackFinal` function. I'll commit changes if it still looks good to you.

🤯 🎉 

I didn't know Storybook already exporting the typing!

However, IMO, using `webpack.Configuration` is not much necessary because it's a "user configuration schema" so we need additional guards for always-existing-properties (`webpackConfig.module!.rules!` and `webpackConfig.resolve!.alias!` OR `webpackConfig.module?.rules?` and `webpackConfig.resolve?.alias ?? {}`). I don't have strong opinion about this so please go with your preference 👍 

Speaking for the additional checks for optional properties, yes it still LGTM (or even better!).


> One more _maybe stupid_ question: although I did not have the chance to check it out, I assume the issue is also occuring with Vue 3 since [vue3's `preset.ts` file](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/vue3/preset.ts) seems to be a simple copy of the vue one. Does this imply a whole new pull request or may this be addressed in this one ?

I think both are fine, we would appreciate it if you can include that in this PR but you don't have to. The only difference is [the package name in L4](https://github.com/storybookjs/storybook/blob/next/addons/docs/src/frameworks/vue3/preset.ts#L4) so applying the same fix is not hard.

Reply:
I've added optional chaining right before iterating the `options.presetsList` and applied the same fix to vue3 as well.

Thank's again for your time, really appreciated! 👍